In [16]:
from sklearn import datasets, metrics, neighbors 
from sklearn import model_selection, svm, naive_bayes

import numpy as np

In [17]:
digit_info = datasets.load_digits()
cancer_info = datasets.load_breast_cancer()

In [26]:
X_dig = digit_info.data
y_dig = digit_info.target
X_can = cancer_info.data
y_can = cancer_info.target

In [30]:
print(X_dig)

[[  0.   0.   5. ...,   0.   0.   0.]
 [  0.   0.   0. ...,  10.   0.   0.]
 [  0.   0.   0. ...,  16.   9.   0.]
 ..., 
 [  0.   0.   1. ...,   6.   0.   0.]
 [  0.   0.   2. ...,  12.   0.   0.]
 [  0.   0.  10. ...,  12.   1.   0.]]


In [31]:
print(X_can)

[[  1.79900000e+01   1.03800000e+01   1.22800000e+02 ...,   2.65400000e-01
    4.60100000e-01   1.18900000e-01]
 [  2.05700000e+01   1.77700000e+01   1.32900000e+02 ...,   1.86000000e-01
    2.75000000e-01   8.90200000e-02]
 [  1.96900000e+01   2.12500000e+01   1.30000000e+02 ...,   2.43000000e-01
    3.61300000e-01   8.75800000e-02]
 ..., 
 [  1.66000000e+01   2.80800000e+01   1.08300000e+02 ...,   1.41800000e-01
    2.21800000e-01   7.82000000e-02]
 [  2.06000000e+01   2.93300000e+01   1.40100000e+02 ...,   2.65000000e-01
    4.08700000e-01   1.24000000e-01]
 [  7.76000000e+00   2.45400000e+01   4.79200000e+01 ...,   0.00000000e+00
    2.87100000e-01   7.03900000e-02]]


In [27]:
clf_mult = naive_bayes.MultinomialNB()
clf_gaus = naive_bayes.GaussianNB()
clf_bern = naive_bayes.BernoulliNB()

In [28]:
print('Multinomial estimation ', model_selection.cross_val_score(clf_mult, X_dig, y_dig).mean())
print('Gaussian estimation', model_selection.cross_val_score(clf_gaus, X_dig, y_dig).mean())
print('Bernoulli estimation', model_selection.cross_val_score(clf_bern, X_dig, y_dig).mean())

Multinomial estimation  0.870877148974
Gaussian estimation 0.818600380355
Bernoulli estimation 0.825823650778


In [29]:
print('Multinomial estimation ', model_selection.cross_val_score(clf_mult, X_can, y_can).mean())
print('Gaussian estimation', model_selection.cross_val_score(clf_gaus, X_can, y_can).mean())
print('Bernoulli estimation', model_selection.cross_val_score(clf_bern, X_can, y_can).mean())

Multinomial estimation  0.894579040193
Gaussian estimation 0.936749280609
Bernoulli estimation 0.627420402859


1. Максимальное качество на дата сете breast_cancer:  0.936749280609.
2. Максимальное качество на дата сете digits: 0.870877148974
3. Верны утверждения: (c), (d).